tlf-assemble.Rmd

```diff
@@ -0,0 +1,154 @@
+
+dput(rtf_table_path)
+#> c("intro-ae1.rtf", "intro-ae2.rtf", "intro-ae3.rtf", "intro-ae5.rtf",
+#> "intro-ae7.rtf", "tbl_disp.rtf", "tbl_pop.rtf", "tlf_ae_summary.rtf",
+#> "tlf_base.rtf", "tlf_eff.rtf", "tlf_eff1.rtf", "tlf_eff2.rtf",
+#> "tlf_km.rtf", "tlf_spec_ae.rtf")
+
+dput(rtf_graph_path)
+#> c("intro-ae1.rtf", "intro-ae2.rtf", "intro-ae3.rtf", "intro-ae5.rtf",
+#> "intro-ae7.rtf", "tbl_disp.rtf", "tbl_pop.rtf", "tlf_ae_summary.rtf",
+#> "tlf_base.rtf", "tlf_eff.rtf", "tlf_eff1.rtf", "tlf_eff2.rtf",
+#> "tlf_km.rtf", "tlf_spec_ae.rtf")
+```
+
+The complete list of `.rtf` files are printed in the Console.
+You can then use this list as a starting point to arrange the order of all files.
+
+An alternative way to get the RTF file list is to read from a spreadsheet
+(e.g., using A&R grid) and work from there.
+
+### Define file order and identify page orientation
+
+Next, we should properly define the order of all TLF files,
+and identify each TLF's page orientation (horizontal or vertical).
+
+The R code below constructs an intermediate data frame `db` with three columns.
+Each TLF file has its respective row to indicate
+
+- `table`: Whether it is table or a plot, `TRUE` or `FALSE`
+- `file`: File name
+- `landscape`: Page orientation, `TRUE` or `FALSE`
+
+<!-- Note to authors: please modify the code below following the last chunk, this is just a placeholder -->
+
+```{r}
+db <- data.frame(
+  file = c(
+    "intro-ae1.rtf",
+    "intro-ae2.rtf",
+    "intro-ae3.rtf",
+    "intro-ae5.rtf",
+    "intro-ae7.rtf",
+    "tbl_disp.rtf",
+    "tbl_pop.rtf",
+    "tlf_ae_summary.rtf",
+    "tlf_base.rtf",
+    "tlf_eff.rtf",
+    "tlf_eff1.rtf",
+    "tlf_eff2.rtf",
+    "tlf_km.rtf",
+    "tlf_spec_ae.rtf"
+  ),
+  table = c(rep(TRUE, 3), FALSE, rep(TRUE, 6), FALSE, rep(TRUE, 2), FALSE),
+  landscape = c(rep(FALSE, 13), rep(TRUE, 1)),
+  stringsAsFactors = FALSE
+)
+
+db
+```
+
+<!-- Note to authors: the next section can probably be avoided if you capture both the file name and the full path in different columns in the data frame above, and consequently avoid the manual copy & paste. I guess one reason you did this is you had to convert the paths from Linux to Windows in the original Rmd, although that is not a generic flow. -->
+
+### Construct full paths to RTF files
+
+The R code below constructs file paths from two directories: `outtable` and `outgraph`.
+You can extend the code to more than two directories.
+
+```{r, eval=FALSE, echo=TRUE}
+full_path <- ifelse(
+  db$table,
+  file.path(path$outtable, db$file),
+  file.path(path$outgraph, db$file)
+)
+```
+
+### Combine RTF files into one file
+
+Combine RTF files into one file and saved in the `path$document` directory.
+
+```{r, eval=FALSE, echo=TRUE}
+r2rtf::rtf_assemble(
+  input = full_path,
+  output = file.path(path$document, "output.docx"),
+  landscape = db$landscape
+)
+```
+
+## Using toggle fields
+
+The files are assembled using toggle fields.
```

Comment:
Is toggle an online tool? Sorry I got lost with the usage of toggle... 
Besides, it looks not clear about the purpose of using toggle... It looks the rtfs are assembled after running `rtf_assemble`

Reply:
To be honest, I'm also confused by how things are described here. However, since this is an almost direct transcription of the original documentation, I believe @elong0527 should be able to improve it further.